3-UsingAPIs/Week2/homework/ex2-pokemonApp/index.js

```diff
@@ -22,18 +22,83 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchData(url) {
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error('Http error');
+    }
+    const jsonData = await response.json();
+    return jsonData;
+  } catch (error) {
+    throw error.message;
+  }
 }
 
-function fetchAndPopulatePokemons(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchAndPopulatePokemons() {
+  try {
+    // get data from the API
+    const data = await fetchData('https://pokeapi.co/api/v2/pokemon?limit=151');
+    const pokemonMenu = document.querySelector('select');
+    const img = document.querySelector('img');
+
+    // fill the menu with pokemons names and urls
+    for (let i = 0; i < data.results.length; i++) {
+      const option = document.createElement('option');
+      option.textContent = `${data.results[i].name}`;
+      option.value = `${data.results[i].url}`;
+      pokemonMenu.appendChild(option);
+    }
+
+
+    // on selecting a Pokemon, get its image
+    pokemonMenu.addEventListener('change', async (e) => {
+      const selectedPokemon = e.target.value;
+      img.src = await fetchImage(selectedPokemon);
+    });
+  } catch (error) {
+    return error;
+  }
 }
 
-function fetchImage(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchImage(pokemonUrl) {
+  try {
+    const fetchImage = await fetch(pokemonUrl);
+    const imageJson = await fetchImage.json();
```

Comment:
You have this wonderful reusable `fetchData()` function which takes a `url` as its parameter and returns the parsed JSON data, while checking for network error. Why not use that instead to make your code more DRY?

```suggestion
    const imageJson = await fetchData(pokemonUrl);
```